### PR TITLE
updated tunables for boosting

### PR DESCRIPTION
**Pull request by topepo**
- lightgbm and catboost have smaller default ranges for the learning rate: -3 to -1 / 2 in log10 units.
  - lightgbm, xgboost, catboost, and C5.0 have smaller default ranges for the sampling proportion: 0.5 to 1.0.
  - catboost engine arguments were added for `max_leaves` and `l2_leaf_reg`.

### CODE REVIEW

File: NEWS.md [hunk 1]
```diff
@@ -1,5 +1,10 @@
 # parsnip (development version)
 
+* Updates to some boosting tuning parameter information:
```
Comment (EmilHvitfeldt) [inline NEWS.md]:
```suggestion
* Updates to some boosting tuning parameter information: (#1306)
```
